server/src/main/java/org/elasticsearch/snapshots/RestoreService.java

```diff
@@ -301,6 +301,9 @@ private void startRestore(
         // Make sure that we can restore from this snapshot
         validateSnapshotRestorable(repository.getMetadata(), snapshotInfo);
 
+        // Make sure that we can restore cluster state from this snapshot
+        validateGlobalStateRestorable(request, snapshot, snapshotInfo);
```

Comment:
This is a little confusing in terms of method name and mechanics (IMO).

Could we maybe just move the check into the conditional below

```
if (request.includeGlobalState()) {
```

and not mutate the request object instead?

Reply:
I was thinking about this, however that is not the only usage. there is also one in `getFeatureStatesToRestore` and could be more in future.